addons/mail/tests/discuss/test_discuss_channel.py

```diff
@@ -203,19 +203,41 @@ def test_channel_info_seen(self):
     def test_set_last_seen_message_should_send_notification_only_once(self):
         chat = self.env['discuss.channel'].with_user(self.user_admin).channel_get((self.partner_employee | self.user_admin.partner_id).ids)
         msg_1 = self._add_messages(chat, 'Body1', author=self.user_employee.partner_id)
+        member = chat.channel_member_ids.filtered(lambda m: m.partner_id == self.user_admin.partner_id)
 
         self.env['bus.bus'].sudo().search([]).unlink()
         with self.assertBus(
-            [(self.env.cr.dbname, "discuss.channel", chat.id)],
-            [{
-                "type": "discuss.channel.member/seen",
-                "payload": {
-                    'channel_id': chat.id,
-                    'id': chat.channel_member_ids.filtered(lambda m: m.partner_id == self.user_admin.partner_id).id,
-                    'last_message_id': msg_1.id,
-                    'partner_id': self.user_admin.partner_id.id,
+            [
+                (self.env.cr.dbname, "discuss.channel", chat.id),
+                (self.env.cr.dbname, "res.partner", self.user_admin.partner_id.id)
+            ],
+            [
+                   {
+                    "type": "mail.record/insert",
+                    "payload": {
+                        "ChannelMember": {
+                            "id": member.id,
+                            "lastSeenMessage": {"id": msg_1.id},
+                            "thread": {
+                                "id": chat.id,
+                                "message_unread_counter": 0,
+                                "message_unread_counter_bus_id": self.env['bus.bus'].sudo()._bus_last_id(),
+                                "model": "discuss.channel",
+                                "seen_message_id": msg_1.id
+                            }
+                        },
+                    },
+                },
+                {
+                    "type": "mail.record/insert",
+                    "payload": {
+                        "ChannelMember": {
+                            "id": member.id,
+                            "lastSeenMessage": {"id": msg_1.id},
+                        },
```

Comment:
Aren't these miss-ordered between target and payload? First payload looks related to res.partner, not channel.

Reply:
Yea indeed, the order does not matter to the `assertBus` method. I'll rearrange them to make it clearer.